drivers/apc_modbus.c

```diff
@@ -1027,14 +1041,23 @@ static void _apc_modbus_usb_lib_to_nut(const modbus_usb_device_t *device, USBDev
 	out->Serial = device->serial_str;
 	out->bcdDevice = device->bcd_device;
 
-	snprintf(bus_buf, sizeof(bus_buf), "%03u", device->bus);
+	res = snprintf(bus_buf, sizeof(bus_buf), "%03u", device->bus);
+	if (res < 0 || (size_t)res >= sizeof(bus_buf)) {
+		assert(0);
+	}
 	out->Bus = bus_buf;
 
-	snprintf(device_buf, sizeof(device_buf), "%03u", device->device_address);
+	res = snprintf(device_buf, sizeof(device_buf), "%03u", device->device_address);
+	if (res < 0 || (size_t)res >= sizeof(device_buf)) {
+		assert(0);
```

Comment:
Is this supposed to error out and abort the driver? Typically we use a `fatalx(EXIT_FAILURE, "some explanation")` in these cases.

Reply:
Yes, it is supposed to do that, though maybe not a good idea if we ever decide to define `NDEBUG`. I'll make it a `fatalx`.

Reply:
Updated.